### DISCUSSION TITLE
When is 1.0.0 coming out?

### DISCUSSION THREAD

**Discussion by alexted** [adriangb/di#87] [General]
Hi! I've been following your project for a long time - it looks very interesting and promising. But me, my friends, colleagues, and most people I talk to in the py community agree that the fact that so far has not been released version 1.0.0 rejects the idea of seriously use your library in production. So, my question is, when is v 1.0.0 released already? If there is no clear understanding, then give at least approximate terms - at a glance, or give a roadmap for which it would be clear what is still missing the library for a full-fledged release.

**Comment by adriangb** [comment]
> Hi! I've been following your project for a long time - it looks very interesting and promising. But me, my friends, colleagues, and most people I talk to in the py community agree that the fact that so far has not been released version 1.0.0 rejects the idea of seriously use your library in production.

I'm curious who the people you mention are. Are they using the project? Have you tested the project in your codebase? Although the project has gained a decent number of stars, I haven't seen many issues opened or PRs. Knowing that I don't write perfect code and certainly not perfect documentation I think this is indicative that these stars are interest more than actual use. The project also has no sponsors, which I think is another indicator of no serious production use. I ask this only because these things are important to understand if the API is good, if there are bugs, etc., all things which we'd ideally want to know before making a 1.0 release and are impossible to know if it's just me working in isolation.

> So, my question is, when is v 1.0.0 released already? If there is no clear understanding, then give at least approximate terms - at a glance, or give a roadmap for which it would be clear what is still missing the library for a full-fledged release.

A `1.0.0` SemVer release means little more than a pledge to _try _not to break things, there's no guarantee. With a project like this I would worry more about other aspects of sustainability than a 1.0.0 release.
I will make a 1.0 release if/when some of the following happen:

- I get at least 1 more stable contributor. I think this is _essential_ (even more so than a 1.0 release) to ensure long term project maintainability.
- All of the open issues are resolved. Several of them will likely require breaking changes.
- I see more issues, PRs or other activity that indicates the project is actually being used.

**Reply by alexted** [reply]
Huh, it's a vicious circle - you don't declare the library stable because you don't see a meaningful influx of users, and users are wary of seriously using your library because it's still not positioned as production-ready! This circle must be broken by someone)

As for the people I mentioned earlier - I am a member of several russian communities on Telegram, for example [fastapi_ru](https://t.me/fastapi_ru) - if you are interested, check them out. That is actually where I learned about this library back in my time - the moderator there is a real evangelist of high-class code (He advised to pay attention to this product, calling it "the only sane di-framework that currently exists".).

**Reply by adriangb** [reply]
FastAPI has 50k stars and hundreds of thousands of users, so I do think it's possible to break that cycle the other way without making promises that might be hard to keep. Besides, user interest is just one aspect of a 1.0 release. A bus factor of at least 2 is another one. Lack of known issues that will likely require a breaking change is another.

Regardless, I appreciate the your suggestion and linking me to the groups you found this library from. Those are some very kind words and I am happy that others find this project interesting and useful.
